scene/3d/navigation_agent_3d.cpp

```diff
@@ -377,7 +399,11 @@ void NavigationAgent3D::update_navigation() {
 	}
 
 	if (reload_path) {
-		navigation_path = NavigationServer3D::get_singleton()->map_get_path(agent_parent->get_world_3d()->get_navigation_map(), o, target_location, true, navigable_layers);
+		if (!(map_override == RID())) {
```

Comment:
Maybe equivalent and cleaner?
```suggestion
		if (map_override.is_valid()) {
```

Reply:
Does it work with both when the RID is reset or nullptr? I remember (some time ago) that this did not work properly in godot 4.0 so out of old habit I used that (ugly) workaround.

Reply:
`is_valid()` checks if the ID is 0, which is the one you get with an empty `RID()`. I don't know if it can be `nullptr`, I think not? Unless it's a `RID *` I guess.
But I'm not an expert on this so not sure.

Reply:
I just tested with a build and it seems to work so I give it a try again.